File: scripts/create-zip.py
#!/usr/bin/env python3
import os
import zipfile
import shutil
from pathlib import Path

script_dir = Path(__file__).parent
project_dir = script_dir.parent
dist_dir = project_dir / 'dist'
releases_dir = project_dir / 'releases'
output_zip = releases_dir / 'streamlist-extension.zip'

# Crear directorio releases
releases_dir.mkdir(exist_ok=True)

# Verificar dist
if not dist_dir.exists():
    print(f"❌ Error: carpeta dist no encontrada en {dist_dir}")
    exit(1)

print("📦 Empaquetando extensión como ZIP...")

# Remover ZIP anterior si existe
if output_zip.exists():
    output_zip.unlink()

# Crear ZIP
try:
    with zipfile.ZipFile(output_zip, 'w', zipfile.ZIP_DEFLATED) as zipf:
        for root, dirs, files in os.walk(dist_dir):
            for file in files:
                file_path = Path(root) / file
                arcname = file_path.relative_to(dist_dir.parent)
                zipf.write(file_path, arcname)
    
    zip_size_mb = output_zip.stat().st_size / (1024 * 1024)
    print(f"\n✅ Extensión empaquetada exitosamente!")
    print(f"📦 Ubicación: {output_zip}")
    print(f"📊 Tamaño: {zip_size_mb:.2f} MB")
    print("\n💡 Para instalar:")
    print("1. Descomprime el ZIP")
    print("2. Abre chrome://extensions")
    print("3. Activa 'Modo de desarrollador' (esquina superior derecha)")
    print("4. Click en 'Cargar extensión sin empaquetar'")
    print("5. Selecciona la carpeta descomprimida")
    
except Exception as e:
    print(f"❌ Error al empaquetar: {e}")
    exit(1)

File: scripts/create-zip.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const distDir = path.join(__dirname, '../dist');
const releasesDir = path.join(__dirname, '../releases');
const outputZip = path.join(releasesDir, 'streamlist-extension.zip');

// Crear directorio releases
if (!fs.existsSync(releasesDir)) {
  fs.mkdirSync(releasesDir, { recursive: true });
}

// Verificar dist
if (!fs.existsSync(distDir)) {
  console.error('❌ Error: carpeta dist no encontrada. Ejecuta primero: npm run build');
  process.exit(1);
}

console.log('📦 Empaquetando extensión como ZIP...');

try {
  // Remover ZIP anterior si existe
  if (fs.existsSync(outputZip)) {
    fs.unlinkSync(outputZip);
  }

  // Usar tar para crear ZIP (más compatible)
  const cmd = process.platform === 'win32'
    ? `powershell -Command "Compress-Archive -Path ${distDir} -DestinationPath ${outputZip} -Force"`
    : `zip -r ${outputZip} dist`;

  execSync(cmd, { 
    stdio: 'pipe',
    shell: true,
    cwd: path.dirname(distDir)
  });

  console.log('\n✅ Extensión empaquetada exitosamente!');
  console.log(`📦 Ubicación: ${outputZip}`);
  console.log('\n💡 Para instalar:');
  console.log('1. Descomprime el ZIP');
  console.log('2. Abre chrome://extensions');
  console.log('3. Activa "Modo de desarrollador" (esquina superior derecha)');
  console.log('4. Click en "Cargar extensión sin empaquetar"');
  console.log('5. Selecciona la carpeta descomprimida');
} catch (error) {
  console.error('❌ Error al empaquetar:', error.message);
  process.exit(1);
}
